app/routes.py

```diff
@@ -1,2 +1,234 @@
+
+#-----------------
+#READ ONE (aka GET)
+@tasks_bp.route("/<id>", methods=["GET"])
+def read_one_task(id):
+    task = get_task_from_id(id)
+    request_body = request.get_json()
+
+    if not task.goal_id:
+        return make_response(jsonify({"task" : task.to_dict()}),200)
+        
+    else:
+        task_goal_response = {
+            "id": task.task_id,
+            "goal_id": task.goal_id,
+            "title": task.title,
+            "description": task.description,
+            "is_complete": task.is_complete(),
+        }
+        return jsonify({"task" : task_goal_response})
+        
+@goals_bp.route("/<id>", methods=["GET"])
+def read_one_task(id):
+    goal = get_goal_from_id(id)
+    return make_response(jsonify({"goal" : goal.to_dict()}),200)
+
+@goals_bp.route("/<goal_id>/tasks", methods=["GET"])
+def read_goal_tasks(goal_id):
+    goal = get_goal_from_id(goal_id)
+    goal_tasks_response = []
+
+    for task in goal.tasks:
+        goal_tasks_response.append({
+            "id": task.task_id,
+            "goal_id": task.goal_id,
+            "title": task.title,
+            "description": task.description,
+            "is_complete": task.is_complete(),
+        })
+
+    return (jsonify({"id": int(goal_id), "title": goal.title, "tasks": goal_tasks_response}), 200)
+    
+#-----------------
+#UPDATE 
+@tasks_bp.route("/<id>", methods=["PUT"]) 
+def update_task(id):
+    task = get_task_from_id(id)
+    request_body = request.get_json()
+    
+    if "title" in request_body:
+        task.title = request_body["title"]
+    if "description" in request_body:
+        task.description = request_body["description"]
+    
+    db.session.commit()
+    
+    return make_response(jsonify({"task" : task.to_dict()}),200)
+
+@goals_bp.route("/<id>", methods=["PUT"]) 
+def update_goal(id):
+    goal = get_goal_from_id(id)
+    request_body = request.get_json()
+    
+    goal.title = request_body["title"]
+
+    db.session.commit()
+    
+    return make_response(jsonify({"goal" : goal.to_dict()}),200)
+
+#-----------------
+#UPDATE --TASK-- COMPLETION STATUS
+@tasks_bp.route("/<id>/<completion_status>", methods=["PATCH"])
+def mark_complete(id, completion_status):
+    task = get_task_from_id(id)
+    #task_dict = {}
+    
+    if completion_status == "mark_complete":
+        task.completed_at = datetime.date
+        
+        #SLACK MESSAGING
+        SLACK_MSG_URL = 'https//slack.com/api/com.postMessage'
+        SLACK_MSG_CHANNEL = 'task-notifications'
+        SLACK_BOT_USERNAME = 'AliesBot'
+        SLACK_TOKEN = os.environ.get("SLACK_TOKEN")
+        slack_msg = "Someone did a thing!"
+        
+        requests.post(SLACK_MSG_URL,{
+            "token": SLACK_TOKEN, 
+            "channel": SLACK_MSG_CHANNEL,
+            "text": slack_msg,
+            "username": SLACK_BOT_USERNAME
+            })
+
+    if completion_status == "mark_incomplete":
+        task.completed_at = None
+
+    #task_dict["task"] = task.to_dict()
```

Comment:
You didn't commit before turning from here.  

Things will look right in your response but won't be saved between requests.